### DISCUSSION TITLE
possibility to emulate tradfri blinds

### DISCUSSION THREAD

**Discussion by comlit** [basilfx/TRADFRI-Hacking#64] [Q&A]
hi
I recently connected my somfy rts blinds to my smart home. I can now control it through http requests. I got some Ikea smart home products already installed in my home, so I wondered if it is possible to emulate a tradfri blind and connect it to my existing work so that in the end it would be possible to control the somfy blind through the Ikea app. From what I saw in this repo it seemed as this would be possible. I would like to hear your thoughts and suggestions on this, since I have got no clue what I'm doing :). thanks

**Comment by basilfx** [comment]
Emulation would mean that you would create some piece of hardware that behaves exactly as the IKEA blinds.

It would be much easier to see how the IKEA blinds work, and how you could hook it up to your existing blinds. So whenever the motor moves, route that signal to your existing blinds. You need to reverse-engineer the existing IKEA solution, to see what signals are sent to the hardware that physically controls the blinds.

Easy answer, hard(er) in reality ;-)

**Comment by PerMejdal** [comment]
I have bought a spare used blind (705.142.61) in IKEA for only ~67€. When I get time, I will make a tear down.

**Comment by MattWestb** [comment]
### That was EXPENCIVE !!!
![IMG_20221014_161426](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/9ef3686c-ec5b-4448-8bdd-a330a05ed9df)
![IMG_20221014_160524](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/d768f8a7-fa22-4c3d-a1e0-5b6a75e9ec39)
![IMG_20221014_160604](https://github.com/basilfx/TRADFRI-Hacking/assets/49618193/9346c4ae-e57b-4850-b6e5-642ad4c25301)
They still using the MG1P module but i think they doing on upgrade to the MG21 and making the same format then they cant producing there own module so long and its expensive and they is buying millions one the MG21 from silbabs so its not expensive for IKEA only one new PCB and firmware from GSDK / EZSP 4/7.2.X.

**Comment by PerMejdal** [comment]
90% rabatt, that is something. I only got 32% rabat, and additional 200 DKK from a coupon.
![IMG_7235](https://github.com/basilfx/TRADFRI-Hacking/assets/13363063/bc1e13e5-20b2-4390-b665-85905d974d58)

**Comment by MattWestb** [comment]
It was 50% off but only the battery was in the package no remote, mounting, changer instructions and so on and it was badly scratched but working OK.
I was asking for guarantee and the rest of the missing things and then was not saying and good things so i was saying its was for the trash but im interested in the Zigbee module and was getting one new price without guarantee.

**Comment by honglihu1986** [comment]
![idlefish-msg-1654849010830](https://github.com/basilfx/TRADFRI-Hacking/assets/39935925/24dc0084-edee-4c60-9fe7-bd83cf2fc3b0)
![Screenshot_20230523-182558_Drive](https://github.com/basilfx/TRADFRI-Hacking/assets/39935925/750ff3f6-ca21-4f48-91d6-a04fe89f9c79)
![L}$U1EH(I{E}2X%R 9HZCV](https://github.com/basilfx/TRADFRI-Hacking/assets/39935925/98e86bf9-2882-42e3-b2cc-6dc23be168f7)

**Comment by MattWestb** [comment]
@honglihu1986 You must have some connection problem with the MG1P chip then it cant being debug locked (as MG21) only writ protecting flash pages.
Did you disconnect the motor and the battery so they cant interference with the debug connection ?
 Here is the `TREDANSEN block-out cellul blind` user data [E2022-UD.zip](https://github.com/basilfx/TRADFRI-Hacking/files/11550556/E2022-UD.zip) and it shall working with one normal IKEA bootloader plus rolling on extracted OTA file like we was doing with Symfonosk 2.

Also UD is only name is overrided and no hardware parameter so they is using the same motor controller board and software on it = very good. 

Edit: In newer version of commander / Segger libs you also need connecting the reset pin or its not working.

**Comment by honglihu1986** [comment]
Thank you very much for the guidance, I don't understand how to rolling on the extracted OTA file, I flashed this user data into an outlet, Home smart 1 shows a blind, but it can't OTA into a blind, it shows the latest firmware.

**Comment by honglihu1986** [comment]
![Screenshot_20230524-160258_Home smart 1](https://github.com/basilfx/TRADFRI-Hacking/assets/39935925/48e534dc-61e2-424d-90be-f9fdcaf98222)

**Comment by MattWestb** [comment]
Ops its little more advanced.
The OTA file is rapped with metadat that need being removes and you is getting one EBL/GBL file that can being flashed with commander.
I must look for the ext instruction but you can doing it with zigpy.cli.

If you have dumping the main flash of one blind its the easiest way only flashing it on one new module and only need the user data for working OK and can being OTA updated without problems.

**Comment by MattWestb** [comment]
For dumping MG1P modules you can using my old guide that is not complete but is working and explaining most things https://github.com/MattWestb/IKEA-TRADFRI-ICC-A-1-Module/tree/master/Flashing-MG

Edit: also possible doing it with Silabs commander CLI or Seggers J-Flasher.

**Comment by MattWestb** [comment]
By the way one updated firmware for all blinds in in the pipe.
https://csa-iot.org/csa_product/connected-roller-blind/
